framework/doc/content/source/multiapps/QuadraturePointMultiApp.md

```diff
@@ -0,0 +1,26 @@
+# QuadraturePointMultiApp
+
+!syntax description /MultiApps/QuadraturePointMultiApp
+
+## Description
+
+Creates a sub-app at the quadrature points within every element in the parent app, which can be leveraged for
+doing multiscale solves. This object requires no special parameters, but this is
+[block restrictable](BlockRestrictable.md) so that the sub-applications can be restricted to only
+be generated on specified subdomains.
+
+!alert note
+Quadrature points that are shared between elements are collapsed into the same sub-application. Please consider the
```

Comment:
Looking at the source, it looks like you are only looking at interior quadrature points anyway

Reply:
I ll adapt the docstring

Reply:
There are two types of default quadrature rules, `QClough` and `QGauss`. Gaussian quadrature points should never be on the element boundaries (https://en.wikipedia.org/wiki/Gaussian_quadrature). Looking through the source it looks like `QClough` uses linear combinations of Gauss points which I'm guessing are still all on the interior (@roystgnr maybe could confirm).

Even a non-default trapezoidal quadrature would have quadrature points at element boundaries but you are iterating over elements and then asking for the quadrature points from that element ... so you would have duplicate points in your `_positions` vector in this case. Maybe this is what you mean by "collapsed"?

Anyway given that default quadrature will probably always be Gauss for MOOSE, I think we can delete this comment now especially that you've added the comment below.

Reply:
sounds good. 
By collapsed i mean we remove the duplicates.
Anyhow it s removed now

Reply:
`QClough` points are all interior; it's basically just a Gaussian rule on each subtriangle inside the macroelement.

We do have more than just `QTrap` giving points on element boundaries: `QGaussLobatto`, `QSimpson`, `QNodal`.  We never get any of those as a default quadrature rule out of a libMesh `FEMContext` or anything, though; Clough-Tocher variables are the only things that don't play well enough with Gauss that we have to account for them.

FIN-S used IIRC `QNodal`, so that all the godawful expensive transcendental-function-filled thermochemical models could be evaluated fewer times (sharing results between elements), but I haven't heard of MOOSE apps doing anything like that.